pkg/services/ngalert/models/receivers.go

```diff
@@ -160,23 +189,60 @@ func IntegrationConfigFromType(integrationType string) (IntegrationConfig, error
 	}
 
 	integrationConfig := IntegrationConfig{Type: config.Type, Fields: make(map[string]IntegrationField, len(config.Options))}
+
 	for _, option := range config.Options {
-		integrationConfig.Fields[option.PropertyName] = IntegrationField{
-			Name:   option.PropertyName,
-			Secure: option.Secure,
-		}
+		integrationConfig.Fields[option.PropertyName] = notifierOptionToIntegrationField(option)
 	}
 	return integrationConfig, nil
 }
 
+func notifierOptionToIntegrationField(option channels_config.NotifierOption) IntegrationField {
+	f := IntegrationField{
+		Name:   option.PropertyName,
+		Secure: option.Secure,
+		Fields: make(map[string]IntegrationField, len(option.SubformOptions)),
+	}
+	for _, subformOption := range option.SubformOptions {
+		f.Fields[subformOption.PropertyName] = notifierOptionToIntegrationField(subformOption)
+	}
+	return f
+}
+
 // IsSecureField returns true if the field is both known and marked as secure in the integration configuration.
-func (config *IntegrationConfig) IsSecureField(field string) bool {
-	if config.Fields != nil {
-		if f, ok := config.Fields[field]; ok {
-			return f.Secure
+func (config *IntegrationConfig) IsSecureField(path IntegrationFieldPath) bool {
```

Comment:
I think the better place for this struct would be in notifier/channel_configs. @JacobsonMT wdyt?